parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -611,6 +624,18 @@ public static Statistics toParquetStatistics(
     return formatStats;
   }
 
+  public static void setStatisticsTruncateLength(int statisticsTruncateLength) {
```

Comment:
This is not correct this way. Writer has a constructor argument so you shall not set here statically. I would suggest creating a new `toParquetStatistics` method with this argument instead.

Reply:
Yes, I know that setting it here is a little hacky. The reasons for that are:  1) ParquetMetadataConverter is created as static inside ParquetFileWriter(don't' understand why). ParquetFileWriter has no chance to pass the parameter that it gets from it's contractor.  2) 'toParquetStatistics()' is defined as public static, so the constructor parameter doesn't wor for it. But you suggested rewriting my own toParquetStatistics(). The concern is code duplicate.   

The #2 is less concern that I can refactor the code a bit to reduce the duplicate. Any suggestions for #1? Maybe change metadataConverter to non-static? Concerns?

Reply:
Just created another iteration by changing the metadataConverter to non-static and create my own  'toParquetStatistics'. Please have a look.

Reply:
Sorry for the late response. I didn't want to recommend refactoring the usage of `MetadataConverter` for your change. I've just ment to have the truncateLength as a parameter of `toParquetStatistics`. You may keep the original method header as well so you do not need to rewrite all the calls only the ones are required for this change.

Reply:
Got it. Thanks.